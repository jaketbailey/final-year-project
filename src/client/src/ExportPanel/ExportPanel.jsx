import './ExportPanel.css';

const ExportPanel = (props) => {
    const saveGeoJSON = () => {
        props.geoJSONLink.click();
    }

    const saveGPX = () => {
        console.log(props.gpxLink)
        props.gpxLink.click();
    }

    return (
        <div className="options-panel__container">
            <div className="options-panel">
               <div>
                    <button className="options-panel__button" onClick={saveGeoJSON} >
                         <img src="/img/save-route/geojson-file.svg" alt="Save Route as GeoJSON" title="Save Route as GeoJSON"/>    
                    </button>
                    <button className="options-panel__button" onClick={saveGPX} >
                         <img src="/img/save-route/gpx-file.svg" alt="Save Route as GPX" title="Save Route as GPX"/>    
                    </button> 
                </div> 
            </div>
        </div>
    )
}

export default ExportPanel;