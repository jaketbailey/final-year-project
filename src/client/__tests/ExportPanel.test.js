import { render } from "@testing-library/react";
import ExportPanel from "../src/ExportPanel/ExportPanel";

describe('Export Panel', () => {
    it('Should render successfully', () => {
        const geoJSONLink = '';
        const gpxLink = ''; 
        
        const component = render(
            <ExportPanel 
                geoJSONLink={geoJSONLink} 
                gpxLink={gpxLink}
            />
        );

        expect(component).toMatchSnapshot();
    })
})