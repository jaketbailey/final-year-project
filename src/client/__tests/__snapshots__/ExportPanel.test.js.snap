// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Panel Should render successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="options-panel__container"
      >
        <div
          class="options-panel"
        >
          <div>
            <button
              class="options-panel__button"
            >
              <img
                alt="Save Route as GeoJSON"
                src="/img/save-route/geojson-file.svg"
                title="Save Route as GeoJSON"
              />
            </button>
            <button
              class="options-panel__button"
            >
              <img
                alt="Save Route as GPX"
                src="/img/save-route/gpx-file.svg"
                title="Save Route as GPX"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="options-panel__container"
    >
      <div
        class="options-panel"
      >
        <div>
          <button
            class="options-panel__button"
          >
            <img
              alt="Save Route as GeoJSON"
              src="/img/save-route/geojson-file.svg"
              title="Save Route as GeoJSON"
            />
          </button>
          <button
            class="options-panel__button"
          >
            <img
              alt="Save Route as GPX"
              src="/img/save-route/gpx-file.svg"
              title="Save Route as GPX"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
